src/cls/tabular/cls_tabular.cc

```diff
@@ -2282,6 +2286,367 @@ int hep_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
     return 0;
 }
 
+
+/**
+ * record hello - a "write" method that creates an object
+ *
+ * This method modifies a local object (in this case, by creating it
+ * if it doesn't exist).  We make multiple write calls (write,
+ * setxattr) which are accumulated and applied as an atomic
+ * transaction.
+ */
+static int inittable_group_obj_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
+{
+  // we can write arbitrary stuff to the ceph-osd debug log.  each log
+  // message is accompanied by an integer log level.  smaller is
+  // "louder".  how much of this makes it into the log is controlled
+  // by the debug_cls option on the ceph-osd, similar to how other log
+  // levels are controlled.  this message, at level 20, will generally
+  // not be seen by anyone unless debug_cls is set at 20 or higher.
+  CLS_LOG(20, "in inittable_group_obj_query_op");
+
+  // see if the input data from the client matches what this method
+  // expects to receive.  your class can fill this buffer with what it
+  // wants.
+  if (in->length() > 100)
+    return -EINVAL;
+
+  bool skipCreate=false;
```

Comment:
i think we can remove create obj if does not already exist.

can you check what is the hctx if this method is executing on a non-existing obj, not sure.